src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -114,7 +115,126 @@ def _bulk_create_with_retry(
             assert isinstance(last_seen_exception, BaseException)
             raise last_seen_exception
 
-    def bulk_record(
+    def _uca_bulk_record(
```

Comment:
it seems like we are copying a lot of code here, is it possible to have the feature flag mechanism within the context of the writes limited instead, basically under 

```python
with writes_limiter.check_write_limits(db_write_keys) as writes_limiter_state:
```

it's a little hard to tell what's the actual change here, but from what I remember where we need to handle this is where we have 

```python
filtered_db_write_keys = use_case_collection.mapping[use_case_id]
```

and changing that to be aware of any of the use cases, is that right?